src/msg/async/AsyncConnection.cc

```diff
@@ -170,6 +170,9 @@ ssize_t AsyncConnection::read_bulk(char *buf, unsigned len)
   nread = cs.read(buf, len);
   if (nread < 0) {
     if (nread == -EAGAIN) {
+      // FIXME: I think this needs to get changed to look like
+      // SimpleMessenger's Pipe::do_recv() EAGAIN handling with
+      // ms_socket_retry_on_eagain
```

Comment:
Waiting for the data in a busy loop makes no sense, there's [e]poll/select for that.

EINTR is different: the data *is* here, however the transfer (from the kernel buffer to the userspace one) has been interrupted, so retrying does make sense.

I think Pipe::do_recv() which should be changed to wait for the data properly (with [e]poll/select)